### DISCUSSION TITLE
Material based platform with user registration, authentication, and gated content

### DISCUSSION THREAD

**Discussion by ben519** [squidfunk/mkdocs-material#3567] [Show and tell]
Thought I'd showcase my ed-tech platform, [Practice Probs](https://www.practiceprobs.com/), here as an example of how you can integrate Material with [Netlify](https://www.netlify.com/) to support user registration and gated content (and in my case, Stripe integration as well for accepting customer payments).

Here's the custom home page with a carousel of my practice problem sets. (Notice the sign up and login buttons in the top right.)
<img width="1792" alt="home" src="https://user-images.githubusercontent.com/723099/153253241-4377678f-7d10-4a47-b860-06cc651f7241.png">

Here's the signup form.
<img width="1792" alt="signup" src="https://user-images.githubusercontent.com/723099/153253326-932df960-9c10-4b6a-b0bf-decf6d524abd.png">

Here's a page that shows gated content in the sidebar. (Notice the lock icons next to gated problem pages.)
<img width="1792" alt="locks" src="https://user-images.githubusercontent.com/723099/153253426-44c3b330-4f97-48db-8333-4bb44182e594.png">

And here's what it looks like when an unauthorized user attempts to visit a gate page. (They get redirected to a 401 page.)
<img width="1792" alt="401" src="https://user-images.githubusercontent.com/723099/153253528-97bc244f-f989-4613-ae77-6fd9f174b4eb.png">

It's not without challenges and limitations, but for anyone else looking to gate content I would highly suggest Netlify. Generally speaking, it's been relatively smooth to set up, coming from a non-software engineer.

Thanks to @squidfunk and various people here for their support.

P.S., I'm hiring Python technical writers if anyone's interested in creating practice problems about XYZ python topic.

**Comment by tahamaddam** [comment]
This looks great as I am working on a similar project (notes/tutorials) 

Would you like to share;

- how did you manage the user registration for gated content?
- how did you deploy the project on Netlify?

**Reply by ben519** [reply]
Thanks. See [this](https://www.netlify.com/blog/2020/07/13/manage-subscriptions-and-protect-content-with-stripe/) and [this](https://www.starfallprojects.co.uk/projects/deploy-host-docs/deploy-mkdocs-material-netlify/#site-setup).

**Reply by tahamaddam** [reply]
Thanks, looking into these.
